Dart/src/com/jetbrains/lang/dart/publicApi/ThirdPartyUtil.java

```diff
@@ -0,0 +1,42 @@
+package com.jetbrains.lang.dart.publicApi;
+
+import com.google.dart.server.AnalysisServerListener;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService;
+import com.jetbrains.lang.dart.ide.actions.DartStyleAction;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * This class is an access point for third party plugins to integrated Dart Plugin features in downstream plugins.
+ */
+@SuppressWarnings("unused")
+public class ThirdPartyUtil {
```

Comment:
I agree that a single "Util" class with unrelated static methods bundled into it isn't good in the long term. It would be better to put the methods for each domain in separate classes. Although this API would consist of Java classes and not a protocol, following the examples of the [Flutter Daemon protocol](https://github.com/flutter/flutter/wiki/The-flutter-daemon-mode) or [Analysis server protocol](https://htmlpreview.github.io/?https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/doc/api.html) seems like a good idea.

Also, if someone is going to take this on, it should we be with the long-term goal of not having an enormous API surface between the Dart plugin and other plugins. Instead we'd review what we have and try to come up with a smaller, stable, less brittle public API with no weird hacks so it will work in the long term. (And in particular, untangling monstrosities like DartVmServiceDebugProcessZ.)

But I don't feel strongly about starting this project now. I won't have time to work on it any time soon, and if nobody is volunteering to work on it, perhaps it is better to wait?